at_secondary/at_secondary_server/lib/src/server/at_secondary_impl.dart

```diff
@@ -144,13 +146,18 @@ class AtSecondaryServerImpl implements AtSecondaryServer {
 
     //Commit Log Compaction
     commitLogCompactionJobInstance = AtCompactionJob(_commitLog);
+    commitLogCompactionStatsInstance = CompactionStats(_commitLog);
     var atCommitLogCompactionConfig = AtCompactionConfig(
         commitLogSizeInKB!,
         commitLogExpiryInDays!,
         commitLogCompactionPercentage!,
         commitLogCompactionFrequencyMins!);
+    commitLogCompactionStatsInstance.initialize();
     await commitLogCompactionJobInstance
         .scheduleCompactionJob(atCommitLogCompactionConfig);
+    commitLogCompactionStatsInstance.calculate();
+    commitLogCompactionStatsInstance.writeStats(commitLogCompactionStatsInstance);
+
```

Comment:
The above code gets called only during the server start-up. I believe the stats will not be collected during every run of compaction.

Reply:
In that case moving this to at_compaction_job will collect stats at every compaction run ?

Reply:
The compaction job runs in at_persistence_secondary_server and stats will in at_secondary_server. You have pass compaction stats information from at_persistence_secondary_server to at_secondary.